src/index.tsx

```diff
@@ -209,10 +211,24 @@ export default class ReactToPrint extends React.Component<IReactToPrintProps> {
     public render() {
         const {
             trigger,
+            onBeforeGetContent
         } = this.props;
 
         return React.cloneElement(trigger(), {
-            onClick: this.handlePrint,
+            onClick: () => {
+                if(onBeforeGetContent) {
+                    const onBeforeGetContentOutput = onBeforeGetContent();
+                    if(onBeforeGetContentOutput && typeof onBeforeGetContentOutput.then === "function") {
+                        onBeforeGetContentOutput.then(() => {
```

Comment:
Nit: can be simplified to: `onBeforeGetContentOutput.then(this.handlePrint)`

Reply:
changed